packages/trie/src/trie/trie.ts

```diff
@@ -94,6 +92,18 @@ export class Trie {
     return new Trie(opts)
   }
 
+  database(db?: DB) {
+    if (db !== undefined) {
+      if (db instanceof CheckpointDB) {
+        throw new Error('Cannot pass in an instance of CheckpointDB')
```

Comment:
I have asked this before but I forgot, why can we not use a CheckpointDB?

Reply:
Ah right, I think my confusion comes from the fact that you pass in a DB, and this actually gets converted into a CheckpointDB later on. (I think this is somewhat confusing in our codebase)